packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -681,7 +681,9 @@ function updateOffscreenComponent(
 
   if (
     nextProps.mode === 'hidden' ||
-    (enableLegacyHidden && nextProps.mode === 'unstable-defer-without-hiding')
+    (enableLegacyHidden &&
+      nextProps.mode === 'unstable-defer-without-hiding') ||
+    workInProgress.stateNode._visibility & OffscreenDetached
```

Comment:
Add TODO to not read from instance once we implement state updates properly in the `attach` PR.

Reply:
> Add TODO to not read from instance once we implement state updates properly in the `attach` PR.